plugins/factory-plugin/src/workspace-projects-manager.ts

```diff
@@ -66,70 +70,150 @@ export class WorkspaceProjectsManager {
     async startSyncWorkspaceProjects() {
         const gitConfigPattern = '**/.git/{HEAD,config}';
         const gitConfigWatcher = theia.workspace.createFileSystemWatcher(gitConfigPattern);
-        gitConfigWatcher.onDidCreate(uri => this.updateOrCreateGitProjectInWorkspace(git.getGitRootFolder(uri.path)));
-        gitConfigWatcher.onDidChange(uri => this.updateOrCreateGitProjectInWorkspace(git.getGitRootFolder(uri.path)));
-        gitConfigWatcher.onDidDelete(uri => this.deleteGitProjectInWorkspace(git.getGitRootFolder(uri.path)));
+        gitConfigWatcher.onDidCreate(uri => this.updateOrCreateProjectInWorkspace(git.getGitRootFolder(uri.path)));
```

Comment:
onDidCreate, onDidChange, onDidDelete returns Disposable, push them to the plugin context please.

Reply:
thanks!

Reply:
They will be disposed when the watcher gets disposed